flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamMultipleInputProcessorFactory.java

```diff
@@ -143,7 +150,20 @@ else if (configuredInput instanceof StreamConfig.SourceInputConfig) {
 				jobConfig
 			);
 
-			inputs = selectableSortingInputs.getSortingInputs();
+			StreamTaskInput<?>[] sortedInputs = selectableSortingInputs.getSortedInputs();
+			StreamTaskInput<?>[] passThroughInputs = selectableSortingInputs.getPassThroughInputs();
+			int sortedIndex = 0;
+			int passThroughIndex = 0;
+			for (int i = 0; i < inputs.length; i++) {
+				if (requiresSorting(inputConfigs[i])) {
+					inputs[i] = sortedInputs[sortedIndex];
+					sortedIndex++;
+				} else {
+					inputs[i] = passThroughInputs[passThroughIndex];
+					passThroughIndex++;
+				}
+			}
+			inputs = selectableSortingInputs.getSortedInputs();
```

Comment:
This line seems to be undoing what you did before, right?

Reply:
Seems like a leftover line but that can lead to bugs.

Reply:
Yes, it's a leftover. 🙈